### PR TITLE
Add the ability to tag users

### PR DESCRIPTION
**Pull request by laserlemon**
This allows us to customize our monthly and weekly reports, scoping them to a subset of users based on tag. Tags could include project team or department.

## Demo

demo

## To-Do

- [x] Write unit tests for `User.tags`
- [x] Write unit tests for `User.with_tags`
- [x] Write a feature test for saving tags

### CODE REVIEW

File: lib/tasks/hourglass.rake [hunk 1]
```diff
@@ -44,15 +44,17 @@ namespace :hourglass do
   task :monthly_report, [:date] => :environment do |_, args|
     Rails.logger.silence do
       date = args[:date] ? Date.parse(args[:date]) : Date.current.last_month
-      puts GenerateReport.call!(range: date.all_month).output
+      tags = (ENV["TAGS"] || ENV["TAG"]).to_s.split(",")
```
Comment (jasonroelofs) [inline lib/tasks/hourglass.rake]: These tasks are now using both rake variables and environment variables. Should we choose one or the other here to stay consistent?

**Comment by jasonroelofs** [comment]
Looks good! I take it you moved from Chosen to Select2 due to Chosen's move into maintenance mode?

If you're interested, there is one small bug that I've been meaning to fix. The caching of the Team name from Harvest does not properly update when you change the Team's project (aka, go from the Q1 project to the Q2 project).

**Comment by laserlemon** [comment]
The move to Select2 was really because of its ability to add custom tags on the fly, which is pretty great. Chosen can't do it. But yeah, it's a bonus that Select2 is now better supported than Chosen!

I can take a look at the bug you mentioned in a separate PR.

File: lib/tasks/hourglass.rake (hunk 1 above)
Comment (laserlemon) [inline lib/tasks/hourglass.rake]: Yes, probably! I'll move to all ENV as it seems more flexible and clear.

Comment (jasonroelofs) [inline lib/tasks/hourglass.rake]: Sounds good. While I like the concept of rake variables, in practice the syntax and usage just feels awkward. ENV variables are fine with me.

**Pull request merged**